mlflow/tracing/utils/otlp.py

```diff
@@ -0,0 +1,61 @@
+import os
+from typing import Optional
+
+from opentelemetry.sdk.trace.export import SpanExporter
+
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import RESOURCE_DOES_NOT_EXIST
+
+
+def should_use_otlp_exporter() -> bool:
+    return _get_otlp_endpoint() is not None
+
+
+def get_otlp_exporter() -> SpanExporter:
+    """
+    Get the OTLP exporter based on the configured protocol.
+    """
+    endpoint, protocol = _get_otlp_endpoint(), _get_otlp_protocol()
+    if protocol == "grpc":
+        try:
+            from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
+        except ImportError:
+            raise MlflowException(
+                "gRPC OTLP exporter is not available. Please install the required dependency by "
+                "running `pip install opentelemetry-exporter-otlp-proto-grpc`.",
+                error_code=RESOURCE_DOES_NOT_EXIST,
+            )
+
+        return OTLPSpanExporter(endpoint=endpoint)
+    elif protocol == "http/protobuf":
+        try:
+            from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+        except ImportError as e:
+            raise MlflowException(
+                "HTTP OTLP exporter is not available. Please install the required dependency by "
+                "running `pip install opentelemetry-exporter-otlp-proto-http`.",
+                error_code=RESOURCE_DOES_NOT_EXIST,
+            ) from e
+
+        return OTLPSpanExporter(endpoint=endpoint)
+    else:
+        raise MlflowException.invalid_parameter_value(
+            f"Unsupported OTLP protocol '{protocol}' is configured. Please set "
+            "OTEL_EXPORTER_OTLP_TRACES_PROTOCOL env variable to either 'grpc' or 'http/protobuf'."
+        )
+
+
+def _get_otlp_endpoint() -> Optional[str]:
+    """
+    Get the OTLP endpoint from the environment variables.
+    Ref: https://opentelemetry.io/docs/languages/sdk-configuration/otlp-exporter/#endpoint-configuration
+    """
+    return os.environ.get(
+        "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT", os.environ.get("OTEL_EXPORTER_OTLP_ENDPOINT", None)
+    )
+
+
+def _get_otlp_protocol() -> str:
+    return os.environ.get(
+        "OTEL_EXPORTER_OTLP_TRACES_PROTOCOL", os.environ.get("OTEL_EXPORTER_OPTL_PROTOCOL", "grpc")
+    )
```

Comment:
```suggestion
    return os.environ.get(
        "OTEL_EXPORTER_OTLP_TRACES_PROTOCOL"
    ) or os.environ.get("OTEL_EXPORTER_OPTL_PROTOCOL", "grpc")
```

nit: to skip `os.environ.get("OTEL_EXPORTER_OPTL_PROTOCOL", "grpc")` if `OTEL_EXPORTER_OTLP_TRACES_PROTOCOL` is available.

Reply:
Wow, I thought it is lazy eval. Python never disappoints me lol.